node/relay_handler.js

```diff
@@ -168,12 +169,29 @@ RelayHandler.prototype.handleRequest = function handleRequest(req, buildRes) {
         return;
     }
     rereq = new RelayRequest(self.channel, req, buildRes);
-    self.reqs[req.id] = rereq;
+    self.sockets[req.remoteAddr].reqs[req.id] = rereq;
     rereq.finishEvent.on(rereqFinished);
     rereq.createOutRequest();
     function rereqFinished() {
-        delete self.reqs[req.id];
+        delete self.sockets[req.remoteAddr].reqs[req.id];
     }
 };
 
+RelayHandler.prototype.getReq = function getReq(req) {
+    var self = this;
+
+    var socket = self.sockets[req.remoteAddr];
+    if (!socket) {
```

Comment:
Doing a set in get. It is perfect for the context being called, but potential an issue for reuse. May be better to put it into set. 

This comment is no blocking.